### DISCUSSION TITLE
Who is the copyright holder?

### DISCUSSION THREAD

**Discussion by markxnelson** [kiali/kiali#4252] [General]
Hi,
I am trying to work out who the copyright holder is for Kiali.  I can't find any copyright notices.  Are you able to confirm who it is? 
Thanks

**Comment by xeviknal** [comment]
Hi Mark,

the whole repo is under [Apache License 2.0](https://github.com/kiali/kiali/blob/master/LICENSE).
I am not sure whether this answer your question though.

Could I ask you the rationale behind this question?
Thanks

**Reply by markxnelson** [reply]
Hi, thanks for your reply.  We are looking at using this code, and our third party approval processes require me to identify who the copyright holder is.  Notice in the very start of the license it says:

      "Licensor" shall mean the copyright owner or entity authorized by the copyright owner that is granting the License.

So it is a natural question to ask who is that copyright holder.  Many, I would say most, k8s/cncf-related repos identify the copyright holder.

**Comment by lucasponce** [comment]
Note that the Kiali "brand" and use was registered initially by Red Hat Inc when project started.
I guess you were asking about that.

But the project works under the Apache License 2.0 as @xeviknal has commented.

**Comment by dneary** [comment]
Absent any specific license with contributors, the developers who make changes maintain their copyrights. That means that your best source for copyright holders is the detailed changelog/git history. It is not uncommon for employers to assert copyright on contributions made by employees on company time, this may be the case for some of the contributions. I am not sure if that is the case of Red Hat employees working on Kiali - typically it is not.

**Comment by jshaughn** [comment]
So, a little bit of follow-up here,  The answer above, by @dneary, is the answer.  The developers basically maintain the copyrights and so there is no single copyright holder for Kiali.  It is possible, if it ever became an issue, that an employing company may assert copyright ownership for its employee.

Overall, handling of the code is dictated by the [license](https://github.com/kiali/kiali/blob/master/LICENSE), which for Kiali is the Apache License 2.0.

Some paraphrased comments from our legal dept:

_It is not required to include a per-source-file license and copyright notice._

_It has been recommended more recently to use a per-source-file SPDX-License-Identifier statements (rather than traditional wordier license notices).  It can have some benefits if users are interested in detailed license data and the use of tools to collect such data. In the Apache case, the SPDX notation would be:_

`SPDX-License-Identifier: Apache-2.0`

**Comment by markxnelson** [comment]
Great, thanks a lot for the followup and clarification.  Most appreciated.
